examples/dixonsolve.C

```diff
@@ -0,0 +1,286 @@
+/* Copyright (C) The LinBox group
+ * ========LICENCE========
+ * This file is part of the library LinBox.
+ *
+ * LinBox is free software: you can redistribute it and/or modify
+ * it under the terms of the  GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+/**\file examples/dixonsolve.C
+ @example examples/dixonsolve.C
+ @author Jean-Guillaume.Dumas@univ-grenoble-alpes.fr
+ @author Romain.Lebreton@umontpellier.fr
+ * \brief Dixon System Solving via Lifting using dense LU or sparse LU
+ * \ingroup examples
+ */
+#include <iostream>
+#include <omp.h>
+
+#include "linbox/algorithms/rational-solver.h"
+#include "linbox/solutions/methods.h"
+#include "linbox/solutions/solve.h"
+#include "linbox/util/args-parser.h"
+#include "linbox/util/error.h"
+#include "linbox/util/matrix-stream.h"
+
+// DenseLU
+#include "linbox/matrix/dense-matrix.h"
+
+// SparseElim
+#include "linbox/matrix/sparse-matrix.h"
+
+using namespace LinBox;
+typedef Givaro::ZRing<Givaro::Integer> Ints;
+typedef DenseVector<Ints> ZVector;
+
+
```

Comment:
This constant prime iterator should be added in  `randiter/random-prime.h` which already has some specialization of prime iterators such as the prime sequence adaptor.
Let's avoid to write utility functions in example files, especially when it is shared by 2 files!

Reply:
ok. done.